### PR TITLE
Allow namespace ignore in user-events

### PR DESCRIPTION
**Pull request by selfxp**
## Description
Some namespaces, such as monitoring or canary services will generate activations with a high cardinality that will increase the storage footprint for Prometheus.
The user-events module should have a configuration for filtering out noisy namespaces.

## Related issue and scope

- [x] I opened an issue to propose and discuss this change (#4667)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [x] Monitoring
- [x] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [x] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [x] I added tests to cover my changes.
- [x] My changes require further changes to the documentation.
- [x] I updated the documentation where necessary.

### CODE REVIEW

File: core/monitoring/user-events/src/main/scala/org/apache/openwhisk/core/monitoring/metrics/EventConsumer.scala [hunk 1]
```diff
@@ -94,6 +95,9 @@ case class EventConsumer(settings: ConsumerSettings[String, String], recorders:
   private def processEvent(value: String): Unit = {
     EventMessage
       .parse(value)
+      .filter { e =>
+        !metricConfig.ignoredNamespaces.contains(e.namespace)
+      }
```
Comment (chetanmeh) [inline core/monitoring/user-events/src/main/scala/org/apache/openwhisk/core/monitoring/metrics/EventConsumer.scala]: Thinking about the ignore support again - May be we should still update the Global metrics so as to account for presence of such activations by ignored namespaces.

Further even having a namespace level metric (which drops the specific action name) would be helpful to still see overall aggregated metric for that namespace. It also helps to confirm if the namespace is not getting abused for some reason (say key leak!)

Comment (selfxp) [reply core/monitoring/user-events/src/main/scala/org/apache/openwhisk/core/monitoring/metrics/EventConsumer.scala]: Makes sense @chetanmeh. Added a new metric for total namespace activations in both `PrometheusRecorder` and `KamonRecorder`

File: core/monitoring/user-events/src/main/scala/org/apache/openwhisk/core/monitoring/metrics/KamonRecorder.scala [hunk 2]
```diff
@@ -87,16 +89,32 @@ object KamonRecorder extends MetricRecorder with KamonMetricNames with SLF4JLogg
         `actionName` -> action,
         `actionKind` -> kind,
         `actionMemory` -> memory)
+    private val namespaceActivationsTags =
+      Map(
+        `actionNamespace` -> namespace,
+        `initiatorNamespace` -> initiator,
+        `actionKind` -> kind,
```
Comment (chetanmeh) [inline core/monitoring/user-events/src/main/scala/org/apache/openwhisk/core/monitoring/metrics/KamonRecorder.scala]: For namespace level metrics we can drop the `actionKind` and `actionMemory` tags

File: core/monitoring/user-events/src/main/scala/org/apache/openwhisk/core/monitoring/metrics/OpenWhiskEvents.scala [hunk 3]
```diff
@@ -33,7 +33,7 @@ import scala.concurrent.{ExecutionContext, Future}
 
 object OpenWhiskEvents extends SLF4JLogging {
 
-  case class MetricConfig(port: Int, enableKamon: Boolean)
+  case class MetricConfig(port: Int, enableKamon: Boolean, ignoredNamespaces: List[String])
```
Comment (chetanmeh) [inline core/monitoring/user-events/src/main/scala/org/apache/openwhisk/core/monitoring/metrics/OpenWhiskEvents.scala]: Better to make it a `Set`

**Pull request merged**